security/pkg/pki/ca/ca.go

```diff
@@ -64,7 +64,7 @@ const (
 // CertificateAuthority contains methods to be supported by a CA.
 type CertificateAuthority interface {
 	// Sign generates a certificate for a workload or CA, from the given CSR and TTL.
-	Sign(csrPEM []byte, ttl time.Duration, forCA bool) ([]byte, error)
+	Sign(csrPEM []byte, subjectIDs []string, ttl time.Duration, forCA bool) ([]byte, error)
```

Comment:
1. this interface isn't exactly aligned with [istio-ca](https://github.com/istio/istio/blob/collab-gcp-identity/security/proto/ca/v1alpha1/istioca.proto#L6), which means nodeagent needs some adapters; I'm fine to this approach, just to confirm
2. looks like ```subjectIDs``` is required field here, while [istio-ca](https://github.com/istio/istio/blob/collab-gcp-identity/security/proto/ca/v1alpha1/istioca.proto#L6) put it(and ttl) as optional, it will use token(```authorization``` header in request) as subjectID(identity), could cidateal provide same support ? so that two interfaces would better aligned and nodeagent don't need to do special handling.